packages/plugin-ext/src/main/node/resolvers/plugin-local-file-resolver.ts

```diff
@@ -0,0 +1,56 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { PluginDeployerResolver, PluginDeployerResolverContext } from '../../../common/plugin-protocol';
+import { injectable } from 'inversify';
+import * as fs from 'fs';
+import * as path from 'path';
+import { FileUri } from '@theia/core/lib/node';
+import URI from '@theia/core/lib/common/uri';
+
+@injectable()
+export class LocalFilePluginDeployerResolver implements PluginDeployerResolver {
```

Comment:
I'm not sure I like that, it's mostly a copy of `LocalDirectoryPluginDeployerResolver`. Maybe someone has an idea on how to improve that.

Reply:
I unified the common logic in ea032a21c83f7b062cf98b2b5f3f38811f5e0248.